packages/pwa/app/pwa-kit-config.json

```diff
@@ -0,0 +1,12 @@
+{
+    "l10n": {
+        "supportedLocales": [
+            "en-GB",
+            "fr-FR",
+            "it-IT",
+            "zh-CN",
+            "ja-JP"
+        ],
+        "defaultLocale": "en-GB"
+    }
```

Comment:
In this [comment in another PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/152#discussion_r718903023), you've mentioned about having a single configuration file for all, and also preferably in json format for schema validation.

Single configuration file? I think that's a good idea for developers needing to know only a single place.

JSON format? I prefer it in JS format instead because:
- like you said, we need a way to call `defineMessages`
- JS also allows us to add code comments

How do other frameworks store their config files? [Vue](https://cli.vuejs.org/config/) and [Next.js](https://nextjs.org/docs/api-reference/next.config.js/introduction) have them in js format.

Perhaps we can have our config in js and then automatically serialize it as a json for easier validation. After all, `defineMessages()` is actually like a [pass-through](https://formatjs.io/docs/intl/#definemessagesdefinemessage)... the output of it is the input object. Its only purpose is to let the static analyzer know which part of the code has the translations.

Reply:
As discuss in person, define messages shouldn't been used in the configuration. As we are not only setting supported locales but setting their definitions as well, kind of double duty. What we'll probably do is have a larger set of locale description definitions provided by the sdk, so the end user doesn't have to supply their own thus maying the number of steps for adding local support easier. We'll work of a BCP 47 subset list that will cover most locale descriptions. If there is the case where they want to support their own locale that isn't in the list, they should be able to add their own definition